### PR TITLE
Example app update

### PR DESCRIPTION
**Pull request by Benbentwo**
## what
* Update workflow filenames and layout to follow new layout of:
  * `reusable` for shared workflows
  * One file per github event
* Extracted all configuration to a `.github/config/app.yaml`
* Action for deploying lambda no longer runs `gh run workflow` replaced with action for dispatch that can wait for success
* Readme Update and modernization
* `.github/config/app.yaml` contains `runs-on-json` which supports passing in how you want to run the workflows

## why
* Example apps need to be quick to be configured for a new user (`.github/config/app.yaml`)
* Files are more clear on where or what is being called by the triggering github event - provides clarity when debugging.
* Synchronous wait for stack deploy means you know your lambda is updated
* Readme Docs template now supports two different types, Cloudposse Docs (if repo lives in cloudposse) or User Docs (when the use this template button is used)

### CODE REVIEW

File: README.yaml [hunk 1]
```diff
@@ -71,4 +101,4 @@ contributors:
   - name: "Jeremy White"
     github: "dudymas"
   - name: "Benjamin Smith"
-    github: "benbentwo"
\ No newline at end of file
+    github: "benbentwo"
```
Comment (milldr) [inline README.yaml:104]: fyi this block doesnt do anything with the latest build-harness readme

File: .github/workflows/main-branch.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  push:
+    branches:
+    - main
+#    paths:
+#    - hello-world-python/**
+#    - .github/workflows/lambda-hello-world-python.yaml
+
+jobs:
+  config:
+    runs-on: self-hosted
```
Comment (milldr) [inline .github/workflows/main-branch.yaml:13]: we've swapped to different self-hosted runner labels internally. We'll very likely need to install additional requirements to work on the new runner set (and so will customers once they start deploying the new component)

Comment (milldr) [reply .github/workflows/main-branch.yaml:13]: `self-hosted-default` for example. Fully list here: [internal slack link](https://cloudposse.slack.com/archives/CA4TC65HS/p1702441032321329)

File: README.yaml (hunk 1 above)
Comment (Benbentwo) [reply README.yaml:104]: This readme isn't generated by the docs folder template

**Review by milldr — APPROVED** [review]

**Pull request merged**